x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/TraceSummary/HttpInfo.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiToolTip, EuiBadge } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import styled from 'styled-components';
+import { idx } from '@kbn/elastic-idx/target';
+import { palettes } from '@elastic/eui';
+import { Transaction } from '../../../../../../typings/es_schemas/ui/Transaction';
+import { units, px, truncate, unit } from '../../../../../style/variables';
+
+const statusColors = {
+  success: palettes.euiPaletteForStatus.colors[0],
+  warning: palettes.euiPaletteForStatus.colors[4],
+  error: palettes.euiPaletteForStatus.colors[9]
+};
+
+function getStatusColor(status: number) {
+  const colors: { [key: string]: string } = {
+    2: statusColors.success,
+    3: statusColors.warning,
+    4: statusColors.error,
+    5: statusColors.error
+  };
+
+  return colors[status.toString().substr(0, 1)] || 'default';
+}
+
+interface HttpStatusBadgeProps {
+  status?: number;
+}
+function HttpStatusBadge({ status }: HttpStatusBadgeProps) {
+  if (!status) {
```

Comment:
I usually prefer to handle this outside of the component, and make this prop required. Often the parent needs to deal with layout changes if the component should not render anything as well.

Reply:
Fixed in 03f58bfcc6.